[댓글 기능] 슬로스 미션 제출합니다

soojinroh opened the pull request.

parker1609 committed:
[게시글 생성/조회/수정/삭제] 코맥 미션 제출합니다. (#31)

* docs : README.md 작성

* feat: 게시글 작성 페이지 이동 구현

* index.html, article-edit.html 파일을 templates 폴더로 이동함

* AriticleController.java: /writing GET요청 처리

Squashed commit of the following:

commit 7fa33210c68665351585da9f8fbc4e2117248a2e
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Tue Jul 9 17:26:55 2019 +0900

    docs: TODO 추가

commit d8e14dbdd71112723278e2e312ec46582a131d00
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Tue Jul 9 17:26:17 2019 +0900

    feat: 게시글 작성 페이지 이동

* feat: 게시글 작성 및 추가 기능 구현

* feat: 게시글 목록 조회

commit 9361045bbb8cfc607918c5ed4bda4ceee9aece6f
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Wed Jul 10 18:01:03 2019 +0900

    feat: 게시글 목록 조회

* feat: 게시글 조회 기능

Squashed commit of the following:

commit 513b0f9f8dcf8cd2a2810db7d8e82c3efd13b317
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Thu Jul 11 08:42:08 2019 +0900

    feat: 게시글 조회 기능 구현

commit 55b1b0ea2760d83c12b1cc060e2fa0d03e9d8297
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Thu Jul 11 08:29:05 2019 +0900

    chore: 절대경로로 변경

    * 외부 리소스 접근 경로 상대경로에서 절대경로로 변경

* feat: 게시글 수정 기능 구현

* feat: 게시글 삭제 기능 구현

Squashed commit of the following:

commit 44fa3af15a2b1b6da8b8229f06a26c315cfa493c
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 14:11:07 2019 +0900

    docs: TODO.md 추가

commit b01f06e590c2b8b42a40c8da36adb01705bfbf7f
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 14:01:39 2019 +0900

    refactor: html js부분 중복 제거

commit 51101469f093fd305653c6ed63d5be8aaa69ad24
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:52:07 2019 +0900

    fix: article-edit.html 나가기 버튼 오류 수정

commit 23ba3a40a3c427248c4ee8f1bdbf16a1119d1798
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:49:11 2019 +0900

    feat: 게시글 삭제 시 redirect

commit 70d5a2e9232f2baea54bbe3ac554696b89ad9149
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:29:53 2019 +0900

    feat: 게시글 삭제 후 남은 게시글 보여주기

commit 184ddc4acf42506d2f46d48eb4e79b4dae6173ce
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:21:21 2019 +0900

    refactor: 들여쓰기

commit 4e8b30e3dfa72ed4babfba2f4e3ffd9b571f8787
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:16:19 2019 +0900

    feat: 게시글 삭제 기능 구현

commit 5111c6e2d26adec5799a4546a3100a0e3f874835
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 01:43:53 2019 +0900

    refactor: Article 클래스 id 필드 추가

commit 2e71b5f6a5010bbe5cecca9a4f26616f53788450
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 01:42:44 2019 +0900

    chore: html 중복 제거

commit b88d426c35f40c12d1e069b9480789d0c7937563
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Thu Jul 11 16:54:29 2019 +0900

    refactor: UI 버그 수정

    * app.css, index.html 수정

commit 7c80edaf0f4452a5f610487086c466ce0da8292a
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Thu Jul 11 16:49:31 2019 +0900

    docs: 게시글 삭제 기능 TODO 작성

* refactor: InvalidArticleException 로 Article 예외 처리

* refactor: implementation 키워드 사용

* refactor: ArticleNotFoundException 추가

* refactor: 게시글 수정 시 원본 게시글 내용이 변경되도록 변경

* refactor: constructor injection으로 변경

* refactor: toArticle() 메서드 추가

* refactor: find()에서 findById()로 이름 변경

* refactor: 게시글 생성 중복 제거

* refactor: 게시글 생성 및 삭제 중복 제거

* refactor: 게시글 수정이 정상적으로 수행되었는지 확인

* refactor: html 중복 제거

* refactor: 그 외 피드백 적용

Squashed commit of the following:

commit 23351a2f8aa281f72d70d94ad5ed89d9c996316d
Author: CODEMCD <psbum1609@gmail.com>
Date:   Sun Jul 14 13:50:22 2019 +0900

    refactor: id 동기화

commit 65fa1ca46087f5b47b85742cd13c871b5f8e197a
Author: CODEMCD <psbum1609@gmail.com>
Date:   Sun Jul 14 13:02:19 2019 +0900

    refactor: RestController 사용

commit 6ae60f47ecc6ed287621da5499d0f9f8aa43bb2b
Author: CODEMCD <psbum1609@gmail.com>
Date:   Sun Jul 14 12:49:41 2019 +0900

    refactor: Collections을 반환할 때 변경 불가하게 수정"

parker1609 committed:
[게시판 회원 기능] 코맥 미션 제출합니다 (#55)

* docs: Readme 업데이트

* feat: 게시글 생성, 목록 조회 기능 구현

* feat: 게시글 조회 구현

* feat: 공통 Head 분리

* feat: html 중복제거

* feat: 게시글 수정

* feat: 게시글 삭제 구현

* feat: devtools 적용

* refactor: Js 분리

* refactor: 컨트롤러 메서드 이름 수정, autowired 삭제

* refactor: 불필요한 메서드 제거

* feat: Article 제목 검증 추가

* feat: error 페이지 추가

* refactor: 피드백 반영

리다이렉트, 이름 잘못 수정된것 원복, header location 검증 추가

* [게시글 생성/조회/수정/삭제] 제이 미션 제출합니다. (#14)

* docs: Readme 업데이트

* feat: 게시글 생성, 목록 조회 기능 구현

* feat: 게시글 조회 구현

* feat: 공통 Head 분리

* feat: html 중복제거

* feat: 게시글 수정

* feat: 게시글 삭제 구현

* feat: devtools 적용

* refactor: Js 분리

* refactor: 컨트롤러 메서드 이름 수정, autowired 삭제

* refactor: 불필요한 메서드 제거

* feat: Article 제목 검증 추가

* feat: error 페이지 추가

* refactor: 피드백 반영

리다이렉트, 이름 잘못 수정된것 원복, header location 검증 추가

* feat: Jpa repository 적용

* docs: README 업데이트

* feat: 회원 등록 기능

* feat: 유저 조회 기능 구현

Squashed commit of the following:

commit 3026e5c7eb726b65b992756d9ec17a73bec140d4
Author: JunHoPark <pjh374968@gmail.com>
Date:   Tue Jul 16 17:54:01 2019 +0900

    feat: 유저 조회 기능 구현

commit c82f723cc643fd9aed7686d0231ed18cd35924b7
Author: JunHoPark <pjh374968@gmail.com>
Date:   Tue Jul 16 16:53:26 2019 +0900

    fix: Valid 애노테이션 추가

    Validation 에러 수정, @Valid가 없으면 검증이 발동되지 않는다

* feat: 로그인 및 로그아웃 기능 구현

Squashed commit of the following:

commit 3af160545dc0fcdce403342652fd1c9497e60982
Author: JunHoPark <pjh374968@gmail.com>
Date:   Wed Jul 17 15:57:01 2019 +0900

    refactor: html 패키지화 원복

commit abaa172ae7507edcc0a8a21852291f5b216d967a
Author: JunHoPark <pjh374968@gmail.com>
Date:   Wed Jul 17 15:53:36 2019 +0900

    feat: 로그아웃 기능 구현

commit a523c371383247a05158711cdc7a1d1c6cd1c2ff
Author: JunHoPark <pjh374968@gmail.com>
Date:   Wed Jul 17 15:26:37 2019 +0900

    feat: 로그인 기능 구현

commit 010778b87f6995e0d38a29d15b3041529cb43b20
Author: JunHoPark <pjh374968@gmail.com>
Date:   Wed Jul 17 15:25:59 2019 +0900

    chore: JPA 실행 쿼리 보기 설정

commit fcf46a795cfdb3dcdd14a683d93dd1c1ea0817b7
Author: JunHoPark <pjh374968@gmail.com>
Date:   Wed Jul 17 14:48:54 2019 +0900

    feat: 세션 검증 Interceptor 구현

commit 261149e005613cc1fe6ab4ead0485842ff80780c
Author: JunHoPark <pjh374968@gmail.com>
Date:   Wed Jul 17 13:12:06 2019 +0900

    refactor: html 패키지 구조 변경

* feat: 회원 수정 및 탈퇴기능 구현

Squashed commit of the following:

commit d36b93b9f56d5a631168ea59425740e3852b4c23
Author: JunHoPark <pjh374968@gmail.com>
Date:   Wed Jul 17 17:29:53 2019 +0900

    feat: 회원 탈퇴 기능 구현

commit 0a0adfd710c88ca96f77b0a14b71812556c0643d
Author: JunHoPark <pjh374968@gmail.com>
Date:   Wed Jul 17 17:07:59 2019 +0900

    feat: 회원 수정 기능 구현

* refactor: html 중복제거 및 세션적용

* fix: Interceptor에서 /users post요청은 통과

* feat: Error 객체의 정의

Login 관련 에러 처리 객체

* test: 세션관련 인수테스트 작성

* test: 회원관련 인수테스트

* refactor: 들여쓰기 및 사용하지 않는 import 제거

* feat: 회원가입시 이메일 중복 오류 처리

* fix: 회원 가입시 email 중복 테스트 수정

* refactor: UserService 분리

* feat: 회원 수정시 이름 형식 오류 처리

* feat: 회원 가입시 유효하지 않은 입력 처리

* refactor: Article 서비스 패키지 분리

* fix: 유저 리스트에서 다른 유저 정보에 접근하는 오류 처리

* feat: 데이터베이스 구축

* 프로덕션 환경은 mysql, 테스트 환경은 h2를 사용하였다.

* refactor: 삼항 연산자로 변경

* login, logout을 구분하는 if-else문을 삼항 연산자로 변경

* article의 기본 그림을 구분하는 if-else문을 삼항 연산자로 변경

* refactor: 삼항 연산자로 변경

* fix: 게시글 id 접근하는 로직 수정

* fix: 절대 경로로 변경

* refactor: 삼항 연산자로 변경, url 변경

* refactor: fragments 패키지로 분리

* refactor: mysql 데이터베이스 설정

* fix: 이미지 경로 수정

* /static/ 추가

* refactor: 패키지 구조 변경

* refactor: dto 클래스들 이름 변경

* refactor: 클래스 이름 변경

* refactor: JpaRepository로 변경

* refactor: 패키지 구조 변경

* refactor: 문자열 상수화

* refactor: 사용하지 않는 부분 삭제

* test: article 도메인 테스트 구현

* feat: 비밀번호 인코딩 구현

* refactor: login GET 메시지인 경우 그대로 login.html 응답

* test: 회원 수정, 회원 탈퇴 테스트 추가

* refactor: 회원 가입 입력 형식 유효성 검사 변경

* User도메인에서 입력 형식 유효성을 검사하는 것에서 UserRequest Dro클래스에서 검사하는 것으로 변경하였다.

* refactor: 사용하지 않는 부분 삭제

* SignUpException 클래스 삭제

* User 클래스에서 더 이상 유효성 검사를 하지 않기 때문에 관련 테스트 삭제

* fix: 회원 가입시 세션 저장 제거

* refactor: 로그인 오류 처리 수정

* 회원 가입과 같은 방법으로 변경

* refactor: 존재하지 않는 게시물 접근 시 에러 처리

* refactor: 에러 테그 위치 이동

* fix: 존재하지 않는 게시물 url에 접근하면 에러 처리

* refactor: 에러 처리 Validator로 변경

* refactor: 회원 수정 오류 처리

* refactor: 존재하지 않는 게시물 url 접근시 에러 페이지로 이동

* fix: User 테스트 코드 오류 수정

* test: 세션을 이용한 로그인 확인 테스트

* refactor: 피드백 반영

* Article, User 클래스 기본 생성자 private으로 변경

* 테스트: random port 제거

* util패키지 incrytor패키지로 이름 변경

* BCryptorImpl 클래스 PasswordBCryptor 클래스로 이름 변경

* 사용하지 않는 improt문 제거, 들여쓰기 맞춤

indiveloper committed:
init myBlog step3

indiveloper committed:
[FEATURE] Apply Converter Pattern at article

indiveloper committed:
[CHORE] fix wrong words

indiveloper committed:
[REFACTOR] change from ArticleRequest to ArticleDto

indiveloper committed:
[REFACTOR] change from Converter to Assembler

indiveloper committed:
[REFACTOR] ArticleDto id 추가

indiveloper committed:
[REFACTOR] 자바 싱글턴 패턴에서 스프링 빈 등록

gch01410 committed:
[REFACTOR] user 클래스에 ModelMapper 적용

ModelMapper를 이용하여 user entity를 UserResponse, UserRequest로 변환
ModelMapper를 이용하여 UserResponse, UserRequest를 User entity로 변환

indiveloper committed:
[CHORE] server.port 설정 추가

8180으로 server.port 설정

indiveloper committed:
[FEATURE] Comment 클래스 구현

indiveloper committed:
[FEATURE] comment 생성 기능 구현

indiveloper committed:
[FIX] session 값 casting 변경

 session의 user attribute 값을 user에서 userResponse로 변경

indiveloper committed:
[REFACTOR] assembler 제거

ArticleService.java: assembler를 사용하여 articleDto에서 article로 변환하는 과정을 modelMapper를 사용하는 것으로 변경하였다.

indiveloper committed:
[FEATURE] Comment delete 기능 구현

작성자 확인 후 다르면 NotSameAuthorException 발생
Comment가 없으면 CommentNotFoundException 발생

indiveloper committed:
[FEATURE] comment update 기능 구현

indiveloper committed:
[FEATURE] 댓글 수정 및 삭제 버튼 유저 확인

Article.html: 해당 게시글에서 댓글 수정 및 삭제 버튼은 이를 작성한 회원만 보이도록 한다.

indiveloper committed:
[REFACTOR] InterceptorTest 로그인 기능 메소드로 분리

indiveloper committed:
[FEATURE] 로그인 여부에 따른 댓글 작성 Interceptor 경로 및 테스트 추가

indiveloper committed:
[TEST] 로그인하지 않은 경우에 대한 테스트 추가

indiveloper committed:
[REFACTOR] 프론트에 비밀번호 확인 기능 추가

백엔드에서 수행하던 비밀번호 확인 기능을 프론트엔드에서 수행하는 것으로 수정

indiveloper committed:
[FIX] UserController Test code 에러 수정

indiveloper committed:
[TEST] CommentControllerTests 구현

댓글 작성, 수정, 삭제, 조회 테스트 구현

indiveloper committed:
[REFACTOR] auto code formatting

indiveloper committed:
merge

indiveloper committed:
resolve conflict

gracefulBrown reviewed (changes requested):
안녕하세요. 리뷰어 브라운 입니다.
댓글 기능 관련 및 테스트 구현 잘 해주셨습니다 👍 
다만 게시글 삭제 시 댓글이 존재하는 경우 처리 등 게시글과 댓글의 연관관계에 대해서 고민이 조금 더 필요할 것 같아요 :)
몇가지 코멘트 남겼으니 참고해주세요 ^^
궁금한 점 있으면 언제든 DM주세요 :)

indiveloper committed:
[REFACTOR] Comment 삭제 및 수정 유저 확인 기능 서비스로 이동

indiveloper committed:
[REFACTOR] 패키지 이름 변경

service 패키지에서 application 패키지로 이름 변경

indiveloper committed:
[REFACTOR] Comment의 changeContents() 메서드 수정

Controller에서 수행하던 수정 및 삭제를 요청한 유저가 실제 해당 댓글을 작성한 유저인지 확인하는 로직을 Comment 클래스와 Service 클래스로 이동

indiveloper committed:
[REFACTOR] 사용하지 않는 클래스 및 메서드 삭제

indiveloper committed:
[FIX] Comment 수정 성공 테스트 에러 수정

indiveloper committed:
[FIX] Annotation 수정 및 주석 삭제

@Column 삭제 및 @JoinColumn에 (nullable=false) 추가

indiveloper committed:
[REFACTOR] StringUtils.isBlank() 적용

indiveloper committed:
[REFACTOR] Article entity에 User author 필드 추가 및 그에 따른 테스트 수정

indiveloper committed:
[REFACTOR] findUserByEmail을 findById로 수정

indiveloper committed:
[FEATURE] CASCADE annotation 추가

Article과 Comment에 CASCASDE annotation 추가

indiveloper committed:
[REFACTOR] 사용하지 않는 생성자 삭제

indiveloper committed:
[REFACTOR] return하는 List를 Collections.unmodifiableList로 변경

indiveloper committed:
[REFACTOR] Comment class의 isSameAuthor 메서드 기능 수정

indiveloper committed:
[REFACTOR] RequestMapping으로 중복 url 제거

indiveloper committed:
[REFACTOR] 피드백 반영

indiveloper committed:
[REFACTOR] Comment Service Test 작성

Mockito를 적용해 Comment Service Test 작성

indiveloper committed:
[REFACTOR] Article Controller에서 article 수정 페이지 접근 확인 기능 추가

indiveloper committed:
[REFACTOR] ModelMapper 적용

indiveloper committed:
[REFACTOR] 변수명 수정

indiveloper committed:
[REFACTOR] 오타 수정

indiveloper committed:
[REFACTOR] Interceptor 수정

indiveloper committed:
[TEST] ArticleServiceTests 구현

mockito를 이용해서 ArticleServiceTests 구현

indiveloper committed:
[TEST] UserTest test 코드 추가

indiveloper committed:
[REFACTOR] 주석 제거 및 포매팅

gracefulBrown reviewed (approved):
커밋 단위를 리팩토링한 내용으로 잘 나누었군요!
리뷰하기 훨씬 편한것 같네요 👍 
프로덕션 코드 / 테스트 코드 에서 피드백 드린 내용이 잘 반영이 되었네요 💯 
고생많으셨습니다! 머지할게요~
